### DISCUSSION TITLE
How do I add multiple TOTPs to a single account entry?

### DISCUSSION THREAD

**Discussion by Night3890** [keepassxreboot/keepassxc#8796] [Q&A]
Hello, 

I have a single account with multiple TOTPs. How do I add both of them to a single account in KeePassXC (I also use KeePassDX on mobile)? I could not find any information on this in the [Manual](https://keepassxc.org/docs/KeePassXC_UserGuide.html#_adding_totp_to_an_entry).

**Comment by droidmonkey** [comment]
You cannot do this and we won't support it. Make two entries, you can clone the first one and just change the totp information.

**Reply by Night3890** [reply]
@droidmonkey Okay. Why will you not support it? It seems like a valuable feature. Is there some problem that it causes for the application?

**Reply by michaelk83** [reply]
note: clone, not close

**Reply by kud** [reply]
I'm not sure when a service provides multiple TOTP? Never seen that. Well, a workaround is to clone the entry I think.

**Reply by Night3890** [reply]
@kud Do you mean when do services give multiple TOTPs for a single account? Often - I know of Crypto exchanges that have different TOTPs to login and to transfer funds. There are also companies that assign the same username/password to login to multiple spaces, but different TOTPs. Those are just examples that I have witnessed. 

I would still like to know why this would be off the table to add as a KeePassXC feature but maybe there is no answer.

**Reply by droidmonkey** [reply]
It would make it unnecessarily cumbersome to use all our features like copy to clipboard, browser extension, and Auto-Type. This works today with a cloned entry for all three of those use cases.

**Reply by kud** [reply]
> @kud Do you mean when do services give multiple TOTPs for a single account? Often - I know of Crypto exchanges that have different TOTPs to login and to transfer funds. There are also companies that assign the same username/password to login to multiple spaces, but different TOTPs. Those are just examples that I have witnessed.
> 
> I would still like to know why this would be off the table to add as a KeePassXC feature but maybe there is no answer.

Oh I see! Thank you for the explanation. Makes sense.

**Reply by lifequilt** [reply]
I use BOTH microsoft and google authenticator for many apps. I always want a backup service for authentication as one may sometimes go down. I would think that GitHub would understand that allowing multiple TOTP services for 2FA is not only possible (and offered by many organizations) but something they must support.

**Reply by kud** [reply]
I think the best way to achieve this for now is to duplicate the item and use the reference (to have just one password and username, allowing updates for both items at the same time) and have a different OTP strategy.
